examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go

```diff
@@ -0,0 +1,156 @@
+func main() {
+	runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:60:64")
+	runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:59:103")
+
+	checker, err := propchecker.NewChecker("./property/example.prop")
+	if err != nil {
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:64:66")
+		log.Fatal(err)
+	}
+	runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:68:75")
+
+	cpuprofile := flag.String("cpuprofile", "", "write cpu profile to `file`")
+	memprofile := flag.String("memprofile", "", "write memory profile to `file`")
+
+	max := flag.Int("n", 5, "defines the number of messages")
+
+	flag.Parse()
+
+	if *cpuprofile != "" {
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:75:78")
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:74:103")
+		f, err := os.Create(*cpuprofile)
+		if err != nil {
+			runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:78:81")
+			runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:80:84")
+			log.Fatal("could not create CPU profile: ", err)
+		}
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:82:83")
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:85:86")
+		if err := pprof.StartCPUProfile(f); err != nil {
+			runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:83:86")
+			runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:81:103")
+			log.Fatal("could not start CPU profile: ", err)
+		}
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:87:88")
+		runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:84:84")
+		defer pprof.StopCPUProfile()
+	}
+	runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:90:115")
+	runtime.ReportBlockCoverage("../examples/ProducerConsumerInstrument/ProducerConsumerInstrument.go:94:105")
+
+	var msgs = make(chan int)	// channel to send messages
+	var done = make(chan bool)	// channel to control when production is done
+
+	go NewProducer(&msgs, &done).produce(*max)
+
+	go NewConsumer(&msgs).consume()
+
+	fmt.Println(" ", <-done)
+
+	sendings := runtime.NumSendings(msgs)
+	deliveries := runtime.NumDeliveries(msgs)
+	fmt.Println("[main]: Channel sendings = ", sendings)
+	fmt.Println("[main]: Channel deliveries = ", deliveries)
+
+	context := make(map[string]interface{})
+	context["main.a"] = sendings
+	context["main.b"] = deliveries
+
+	result, failures, err := checker.Check(context)
```

Comment:
You shouldn't be using checker here. Type B instrumentation requires using DaraLog statements to reports the values of the variables to the runtime.
Dara has its own checker that it uses.